mesonbuild/backend/ninjabackend.py

```diff
@@ -1261,7 +1261,7 @@ def generate_coverage_legacy_rules(self, gcovr_exe: T.Optional[str], gcovr_versi
     def generate_install(self):
         self.create_install_data_files()
         elem = self.create_phony_target('install', 'CUSTOM_COMMAND', 'PHONY')
-        elem.add_dep('all')
+        elem.add_dep('all-install')
```

Comment:
I guess that if a new target name is added, it should be added to `FORBIDDEN_TARGET_NAMES` in coredata module.